src/provider/controller/Layouter.ts

```diff
@@ -0,0 +1,249 @@
+import {EventEmitter} from 'events';
+
+import {injectable} from 'inversify';
+import {PointTopLeft} from 'openfin/_v2/api/system/point';
+import {Rect} from 'openfin/_v2/api/system/monitor';
+import {Transition, TransitionOptions} from 'openfin/_v2/api/window/transition';
+import {MonitorEvent} from 'openfin/_v2/api/events/system';
+import Bounds from 'openfin/_v2/api/window/bounds';
+import {_Window} from 'openfin/_v2/api/window/window';
+
+interface LayouterConfig {
+    spacing: number;
+    anchor: PointTopLeft;
+    animationTime: number;
+}
+
+export interface LayoutItem {
+    animate(transitions: Transition, options: TransitionOptions): Promise<void>;
+    setTransform(transform: Bounds): Promise<void>;
+    readonly dimensions: Promise<WindowDimensions>;
+    position: PointTopLeft;
+}
+
+export interface LayoutStack {
+    items: LayoutItem[];
+    layoutHeight: number;
+}
+
+export type WindowDimensions = {height: number, width: number};
+
+export enum LayoutEvent {
+    LAYOUT_REQUIRED = 'layout-required'
+}
+
+@injectable()
+export class Layouter {
+    private static INTERNAL_CONFIG: Readonly<LayouterConfig> = {
+        spacing: 10,
+        anchor: {top: 1, left: 1},
+        animationTime: 100
+    };
+
+    private _availableRect!: Required<Rect>;
+
+    public static eventEmitter: EventEmitter = new EventEmitter();
```

Comment:
Are we sure an eventEmitter is the right choice here? We've been using `Signals` for most similar things in this and the other services.